app/controllers/budgets/groups_controller.rb

```diff
@@ -1,5 +1,7 @@
 module Budgets
   class GroupsController < ApplicationController
+    before_action :load_budget
```

Comment:
These before_actions should not be necessary
How about moving the logic to raise the exception to the `show` action?